Fe bug fixes

hicks8989 opened the pull request.

hicks8989 committed:
Refactored Front End Code.

- Created Authentication class instance to handle authentication flow of application.
- Created service files to handle internal api calls to server.
- Created helper files to help with api calls.
- Created PrivateRoute component to handle private routes.
- Updated Home, Login, Register, Router, and App components to use services and Authentication class instance.

hicks8989 committed:
Fixed bugs

- Removed bad props from components.
- Updated handle-response file.
- Updated authentication.service file.
- Updated backend route for getting authenticated user.

hicks8989 committed:
Fixed authentication issue.

- Updated user.service file.
- Updated authentication.service file.

hicks8989 committed:
Update user.js

dbm19 reviewed (approved):
Looks good, I think there's just a small typo in App.js. It imports Router from "./Router", but the real file name is "./router"

hicks8989 committed:
Fixed routing design issue.

- Added query-string to package.json.
- Created feed component to replace internal router.

hicks8989 committed:
Rename router.js to Router.js

hicks8989 committed:
Refactored Code.

- Created Invitation.service file to handle invitation api calls.
- Updated invitation component to use invitation.service file.
- Added getData function to helpers.
- Updated backend server auth api call.
